### ISSUE TITLE
Unable to call ASP types from .NET 4.8 API Application

### ISSUE THREAD

**Issue by bleepzter** [dotnet/aspnet-api-versioning#984]
If you create an app with .NET 4.8 as shown in the `examples/WebApi/OPenApiWebApiExample` you cannot physically reference the ApiVresionRouteConstraint, as well as the AddApiVersioning and the UseWebApi method of the builder because of ambiguities.

Please provide a path to execute the same codebase but without referencing the .net project but using the .NET packages directly.

**Comment by bleepzter** [comment]
This is a huge blocker because in older 4.8 framework implementations the swagger / swashbuckle api does not generate swagger.json therefore making it extremely difficult to consume such apis when you can't generate a client with nswag.

**Comment by commonsensesoftware** [comment]
I'm not sure I follow. What do you mean by _physically reference the `ApiVersionRouteConstraint`_? The type `Asp.Versioning.Routing.ApiVersionRouteConstraint` is public and fully accessible. In fact, that usage is shown directly in the example as:

```c#
var constraintResolver = new DefaultInlineConstraintResolver()
{ 
    ConstraintMap = { ["apiVersion"] = typeof( ApiVersionRouteConstraint ) },
};
```

I don't understand what the _ambiguities_ the `AddApiVersioning` extension method are. `UseWebApi` is part of the OWIN process to bootstrap in ASP.NET Web API.

I'm going to need some more information as well as what specifically is blocking you. The example projects are - well - _examples_. They are only meant to show one possibility of many. If you have a specific repro that you can share, I'm happy to take a peek to help you out.

**Comment by bleepzter** [comment]
For the record - this is how to reproduce the issue:
- Visual Studio 2022
- .NET 4.8

Create a new empty Web (.NET Framework 4.8) application. Bring in the following NuGet packages:
Microsoft.Owin.Host.SystemWeb
Microsoft.AspNet.WebApi
Microsoft.AspNet.WebApi.Client
Microsoft.AspNet.WebApi.Core
Microsoft.AspNet.WebApi.Owin
Microsoft.AspNet.WebApi.Versioning
Microsoft.AspNet.WebApi.Versioning.ApiExplorer
Microsoft.AspNet.WebApi.WebHost
Swashbuckle
Swashbuckle.Core

If you substitute Microsoft.AspNet.WebApi.Versioning with Asp.Versioning you'll see the issues.

So in your project - the Asp namespace comes from the referenced project. If you don't reference your project but reference the NuGet packages instead - there are namespace and type collisions between the Asp.Versioning.Routing NuGet package and the existing  Microsoft.Web.Http.Versioning or System.Web.Http.Routing and Microsoft.Web.Http.Routing.

The "DefaultInlineConstraintResolver" is one place where this ambiguity makes Visual Studio 2022 Pro go nuts.

**Comment by commonsensesoftware** [comment]
I see. That's easy to explain. `Microsoft.AspNet.WebApi.Versioning.*` is EOL. `Asp.Versioning.WebApi.*` is the successor. This has been the case for two major version releases already. Mixing and matching is not supported and will naturally lead to all kind of conflicts and issues.

TL;DR, the details were outline in discussions #807 and #808. I've put up banners on the repo and new packages to help facilitate communication. If you're still confused, I feel your pain and fell on a lot of grenades for the community. The powers that be at MS have simply left us in the dust despite my best efforts make sure that didn't happen. Community kvetching is probably the only thing that will change anything.

The biggest difference in the migration is package and namespace change. From a feature and type perspective, most things are the exactly the same. If you're stuck on a specific part, I'm happy to help walk you through the necessary changes.

**Comment by commonsensesoftware** [comment]
This thread has gone idle and presume the issue was resolved by swapping out the package and versions. If you still need help migrating, feel free to ask more questions or re-open the issue.
